auth/src/main/java/com/firebase/ui/auth/util/data/TaskFailureLogger.java

```diff
@@ -0,0 +1,38 @@
+package com.firebase.ui.auth.util.data;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.firebase.ui.auth.AuthUI;
+import com.firebase.ui.auth.BuildConfig;
+import com.google.android.gms.tasks.OnFailureListener;
+
+public class TaskFailureLogger implements OnFailureListener {
+    private static final boolean LOG;
+
+    private String mTag;
+    private String mMessage;
+
+    static {
+        boolean log;
```

Comment:
Huh this is a neat but also kind of nasty trick.  Did you see this somewhere else?  Reflection in a static block seems like it could really slow down some startup times.

Maybe we should just add a FirebaseUI.setLogLevel method or something?

Reply:
Yeah, I just noticed that some exceptions had sensitive info like emails and whatnot that we shouldn't be logging in production. I'll put this on the TODO list for later to figure out. 👍